plotnine/scales/scale.py

```diff
@@ -624,9 +602,10 @@ def break_info(self, range=None):
             major = major.compress(np.isfinite(major))
             minor = self.get_minor_breaks(major, range)
 
-        major = major.compress(
-            (range[0] <= major) & (major <= range[1]))
         labels = self.get_labels(major)
+        mask = (range[0] <= major) & (major <= range[1])
+        major = [x for x, m in zip(major, mask) if m]
+        labels = [x for x, m in zip(labels, mask) if m]
```

Comment:
Okay, I have failed. I tried to think of a way to restrict the changes to this single location, but the other options are worse. There isn't a good way to represent missing values in the python ecosystem, it would have been helpful.

Also, `labels = self.get_labels(major)` has to use the compressed `labels` so that the labelling in `mizani` can make a good decision especially when dealing with large number ranges.

So change this back closer to your original solution but call the extra parameter a `mask` and add a docstring for
`get_labels`.

Thanks for you patience.